src/python/bot/fuzzers/afl/launcher.py

```diff
@@ -1120,6 +1139,40 @@ def libfuzzerize_corpus(self):
 
     return new_units_generated, new_units_added, corpus_size
 
+  def generate_new_testcase_mutations(self):
```

Comment:
any way we could refactor this so that libFuzzer and AFL can share the same code rather than duplicating logic?

Reply:
Was going to, but they handle it a little differently. AFL adds to the existing input corpus whereas LibFuzzer makes and returns a new one, I thought this would be the cleanest way to deal with that difference. Any ideas on how to handle?

Reply:
how about a general function which takes in an output and and input dir? When calling from libFuzzer, we create the new dir and pass it to the function as the output. for AFL, we just pass it with the same arguments for input and output?

Reply:
Sure. I'll get started on that. The other thing was that I thought it might get messy because for libFuzzer we return the new directory, but we don't need to for AFL. Also, the way that they handle keeping track of used strategies is different so I have to work around that to set the generator strategy variables for each of them.

Reply:
I'll do the refactor and you can let me know how it looks

Reply:
Done. I think it's better like this.